### DISCUSSION TITLE
Running Jellyfin with MPC-HC

### DISCUSSION THREAD

**Discussion by IcePlains** [jellyfin/jellyfin-mpv-shim#335] [Q&A]
Can anyone help me with playback on MPC-HC (Windows) with Jellyfin MPV Shim, so i can use MPC-HC instead of integrated mpv?
I think i need to change conf.json, but it is not working yet.

```
    "mpv_ext": true,
    "mpv_ext_ipc": null,
    "mpv_ext_no_ovr": false,
    "mpv_ext_path": "C:\\Program Files\\MPC-HC\\mpc-hc64.exe",
    "mpv_ext_start": true,
```
When i start Jellyfin MPV Shim, it says:
```
Unrecognized switch(es) found in command line string: 
C:\Program Files\MPC-HC\mpc-hc64.exe --input-default-bindings=yes --input-vo-keyboard=yes --input-media-keys=yes --player-operation-mode=cplayer --osc=no --script=C:\Program Files (x86)\Jellyfin MPV Shim\jellyfin_mpv_shim\mouse.lua --script=C:\Program Files (x86)\Jellyfin MPV Shim\jellyfin_mpv_shim\thumbfast.lua --script=C:\Program Files (x86)\Jellyfin MPV Shim\jellyfin_mpv_shim\trickplay-osc.lua --config=yes --config-dir=C:\Users\User\AppData\Roaming\jellyfin-mpv-shim --idle=yes --input-ipc-server=mpv1111111111111111 --input-terminal=no --terminal=no
Usage: mpc-hc.exe "pathname" [switches]
...
```
Thank you

**Comment by iwalton3** [comment]
This isn't supported and I'm unlikely to add it because it requires major architrctural changes.

PRs are welcome but they must be maintainable.

**Comment by IcePlains** [comment]
Pity and thanks for answering.

How i do it now: in Jellyfin i "Copy Stream URL", and paste it in MPC-HC.
I though i would be able to make it a bit more straightforward.

**Reply by Atmaweaponvi** [reply]
would MPC shim be of any use to you? https://github.com/Trellmor/jellyfin-mpc-shim/releases

**Comment by zotabee** [comment]
I would die to have this too!

On Kodi it's super simple, just need to create and edit an xml file like this : https://kodi.wiki/view/External_players

` <playercorefactory>
  <players>
    <player name="MPC-HC" type="ExternalPlayer" audio="false" video="true">
      <filename>C:\Program Files\MPC-HC\mplayerc.exe</filename>
      <args>"{1}" /fullscreen /close</args>
      <hidexbmc>false</hidexbmc>
      <hideconsole>false</hideconsole>
      <warpcursor>none</warpcursor>
    </player>
  </players>
  <rules action="prepend">
    <rule filetypes="mkv" filename=".*720p.*" player="MPC-HC"/>
  </rules>
 </playercorefactory>`

Pass the args there, MPC-HC doesn't need a lot, just the filename, fullscreen and the close at end args. That's it, pretty simple.

Also just found out : https://features.jellyfin.org/posts/227/external-player-support-desktop

> There is a solution for Windows using TamperMonkey!!!
> 
> https://github.com/tccoin/Jellyfin-Potplayer

**Reply by zotabee** [reply]
Replying to myself but huge thanks to @Atmaweaponvi !!! jellyfin-mpc-shim : https://github.com/Trellmor/jellyfin-mpc-shim
